app/client/cli/keys.go

```diff
@@ -0,0 +1,478 @@
+				keybasePath, err := filepath.Abs(pocketDir + keybaseSuffix)
+				if err != nil {
+					return err
+				}
+				kb, err := keybase.NewKeybase(keybasePath)
+				if err != nil {
+					return err
+				}
+
+				if !nonInteractive {
+					pwd = readPassphrase(pwd)
+				}
+
+				// Determine correct way to export private key
+				var exportString string
+				switch strings.ToLower(exportAs) {
+				case "json":
+					exportString, err = kb.ExportPrivJSON(addrHex, pwd)
+					if err != nil {
+						return err
+					}
+				case "raw":
+					exportString, err = kb.ExportPrivString(addrHex, pwd)
+					if err != nil {
+						return err
+					}
+				default:
+					return fmt.Errorf("invalid export format: got %s, want [raw]/[json]", exportAs)
+				}
+
+				if err := kb.Stop(); err != nil {
+					return err
+				}
+
+				// Write to stdout or file
+				if outputFile == "" {
+					logger.Global.Info().Str("private_key", exportString).Msg("Key exported")
+					return nil
+				}
+
+				return shared.WriteOutput(exportString, outputFile)
+			},
+		},
+	}
+	return cmds
+}
+
+func keysImportCommands() []*cobra.Command {
+	cmds := []*cobra.Command{
+		{
+			Use:     "Import [privateKeyString] [--input_file] [--import_format]",
+			Short:   "Imports a key from a string or from a file",
+			Long:    "Imports [privateKeyString] or from [--input_file] into the keybase, provided it is in the form of [--import_format]",
+			Aliases: []string{"import"},
+			Args:    cobra.MaximumNArgs(1),
+			RunE: func(cmd *cobra.Command, args []string) (err error) {
+				// Get import string
+				var privateKeyString string
+				if len(args) == 1 {
+					privateKeyString = args[0]
+				} else if inputFile != "" {
+					privateKeyString, err = shared.ReadInput(inputFile)
+					if err != nil {
+						return err
+					}
+				} else {
+					return fmt.Errorf("no input file provided")
+				}
+
+				// Open the debug keybase at the specified path
+				pocketDir := strings.TrimSuffix(dataDir, "/")
+				keybasePath, err := filepath.Abs(pocketDir + keybaseSuffix)
+				if err != nil {
+					return err
+				}
+				kb, err := keybase.NewKeybase(keybasePath)
+				if err != nil {
+					return err
+				}
+
+				if !nonInteractive {
+					pwd = readPassphrase(pwd)
+				}
+
+				// Determine correct way to import the private key
+				switch strings.ToLower(importAs) {
+				case "json":
+					kp, err := kb.ImportFromJSON(privateKeyString, pwd)
+					if err != nil {
+						return err
+					}
+					logger.Global.Info().Str("address", kp.GetAddressString()).Msg("Key imported")
+				case "raw":
+					kp, err := kb.ImportFromString(privateKeyString, pwd, hint)
+					if err != nil {
+						return err
+					}
+					logger.Global.Info().Str("address", kp.GetAddressString()).Msg("Key imported")
+				default:
+					return fmt.Errorf("invalid import format: got %s, want [raw]/[json]", exportAs)
+				}
+
+				return kb.Stop()
+			},
+		},
+	}
+	return cmds
+}
+
+func keysSignCommands() []*cobra.Command {
```

Comment:
```suggestion
// TODO: Add the following key signature commands
// 	1. Read an encoded unsigned transaction from `--input_file` and write the signed transaction to `--output_file`
// 	2. Verify the signature of a transaction provided by `--input_file`
func keysSignCommands() []*cobra.Command {
```

Reply:
@Olshansk is the message going to be hex encoded? I will add this tonight

Reply:
Not hex encoded, but a marshalled version of the `Transaction` proto written to a file.

Reply:
> Just trying to figure out what the --input_file will look like, something like this? Or the raw bytes written to a file?

Look at `transaction.proto`:
```proto
message Transaction {
  google.protobuf.Any msg = 1; // CONSOLIDATE: Should be a oneof {} message
  Signature signature = 2; // CONSOLIDATE: should use a shared crypto package type
  string nonce = 3;
}

// REFACTOR: Consolidate with other signature types throughout the codebase
message Signature {
  bytes public_key = 1;
  bytes signature = 2;
}
```

A serialized (i.e. marshalled) version of `Transaction` is what we send over the wire. Someone could theoretically prepare that it for you, file and write it to disk (without the `Signature` piece), so you'd read it from file, unmarshal, update the signature, re-marshal (i.e. serialize), write to file.

> 	Should the VerifyTx command also require a --transaction_file or something like this flag - or should the signature be written to --output_file with the tx as well?

For this, you'd basically read a serialized `transaction.proto` from disk, and check the signature against the provided address (which you'd need to get the public key for).

Does that help? 

---

Note that we'll likely have other flows & use cases in the future, but this will also help in E2E automation in the future.

Reply:
👍🏼  On it ;)